homeassistant/helpers/entity_registry.py

```diff
@@ -616,6 +617,38 @@ def async_update_entity_options(
 
         return new
 
+    @callback
+    def async_migrate_entity_to_new_platform(
+        self,
+        entity_id: str,
+        new_platform: str,
+        *,
+        new_unique_id: str | UndefinedType = UNDEFINED,
+        new_config_entry_id: str | None | UndefinedType = UNDEFINED,
+        new_device_id: str | None | UndefinedType = UNDEFINED,
+    ) -> RegistryEntry:
+        """
+        Migrate entity to new platform.
+
+        This should only be used when an entity needs to be migrated between
+        integrations.
+        """
+        if not self.async_get(entity_id):
+            raise ValueError(f"Entity {entity_id} is not in the entity registry")
+        if (
+            state := self.hass.states.get(entity_id)
+        ) is not None and state.state != STATE_UNKNOWN:
+            raise ValueError("Only entities that haven't been loaded can be migrated")
+
+        old = self.entities[entity_id]
+        new = self.entities[entity_id] = attr.evolve(old, platform=new_platform)
+        return self.async_update_entity(
+            new.entity_id,
+            new_unique_id=new_unique_id,
+            config_entry_id=new_config_entry_id,
```

Comment:
Maybe this is should be required if the entity's config_entry is not None?
Or is it possible that the config entry does not change?

Reply:
I don't see how the `config_entry` does not change if it is not `None` as there is no way to migrate a config entry to a new platform, nor do I see any value in doing so.